src/middlewared/middlewared/etc_files/local/ssh/config.py

```diff
@@ -24,6 +25,16 @@ def generate_ssh_config(middleware):
         if os.path.exists(os.path.join(SSH_CONFIG_PATH, f)):
             os.chmod(os.path.join(SSH_CONFIG_PATH, f), 0o600)
 
+    twofactor_auth = middleware.call_sync('auth.twofactor.config')
+    if twofactor_auth['enabled'] and twofactor_auth['services']['ssh']:
+        with open(USERS_OATH_FILE, 'w') as f:
```

Comment:
So, there is a small security breach here, you create and write the file, only then you chmod it.
That leaves an opportunity for an attacker to read the file before you chmod it.

We should do `with open(os.open(USERS_OATH_FILE, os.O_CREAT | os.O_WRONLY, 0o600), 'w') as f:`

Reply:
I was thinking that we can throw something from `mako` template that will make mako stop generating the file and delete it if it already exists.

Reply:
Good idea

Reply:
raise DeprecatedFile ? : P